### PR TITLE
API for retrieving all info on tutorial

### PR DESCRIPTION
**Pull request by Arcadius19**
Added an API to retrieve the whole tutorial, i.e. all its properties. It is required for an admin interface where the user can modify any tutorial

### CODE REVIEW

File: api/tutorial.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 const tutorialDB = require('../db/tutorial-queries')
 const authCheck = require('../utils/authentication-check')
 
-function createTutorial (req, res, next) {
+function createTutorial (req, res) {
```
Comment (NickWu007) [inline api/tutorial.js]: Is there a reason to change this?

File: routes/index.js [hunk 2]
```diff
@@ -19,6 +19,7 @@ router.get('/api/logout', userAPI.logout)
 router.post('/api/tutorial/create', tutorialAPI.createTutorial)
 router.post('/api/tutorial/update', tutorialAPI.updateTutorial)
 router.post('/api/tutorial/delete', tutorialAPI.removeTutorial)
+router.post('/api/tutorial', tutorialAPI.getTutorial)
```
Comment (NickWu007) [inline routes/index.js]: I would suggest a `GET` for this, such as `GET /api/tutorial/:id`

**Review by NickWu007 — CHANGES_REQUESTED** [review]

File: api/tutorial.js (hunk 1 above)
Comment (Arcadius19) [reply api/tutorial.js]: It's redundant and other functions in the file do not have it.

File: routes/index.js (hunk 2 above)
Comment (Arcadius19) [reply routes/index.js]: I thought so but I followed the way you used for retrieving a user's source code for a particular tutorial as in here:
<https://github.com/links-lang/TryLinks-Server/blob/f2f472490c0718dbfef7bd7d3c5cd0de3ca5bcb5/routes/index.js#L13>
I will change it to GET request though as it is more intuitive.

File: api/tutorial.js (hunk 1 above)
Comment (NickWu007) [reply api/tutorial.js]: My bad. nvm

File: routes/index.js (hunk 2 above)
Comment (NickWu007) [reply routes/index.js]: We should probably change the file api endpoints in the future, thanks for pointing out.

For now let's go with `GET` for the tutorials.

**Review by NickWu007 — APPROVED** [review]

**Pull request merged**